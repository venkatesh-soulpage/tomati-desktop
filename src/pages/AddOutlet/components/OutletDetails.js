import React from "react";
import { Form, Button, Card } from "react-bootstrap";

const OutletDetails = ({ handleStep, values, handleChange, setValues }) => {
  const fileToBase64 = async (file) =>
    new Promise((resolve, reject) => {
      const reader = new FileReader();
      reader.readAsDataURL(file);
      reader.onload = () => resolve(reader.result);
      reader.onerror = (e) => reject(e);
    });
  return (
    <div>
      <Form.Group>
        <Form.Control
          type="text"
          placeholder="Outlet Name"
          value={values.name}
          onChange={handleChange("name")}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Control
          type="text"
          placeholder="Phone Number"
          value={values.phone_number}
          onChange={handleChange("phone_number")}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Control
          type="text"
          placeholder="Address"
          value={values.address}
          onChange={handleChange("address")}
          as="textarea"
          rows={3}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Control
          type="text"
          placeholder="Description"
          value={values.description}
          onChange={handleChange("description")}
          as="textarea"
          rows={3}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Control
          id="coverImage"
          label="Custom file input"
          type="file"
          custom
          className="d-none"
          value={values.cover_image}
          onChange={async (e) => {
            console.log(e.target.files[0].name);
            const url = await fileToBase64(e.target.files[0]);
            console.log(url);
            // setValues({ ...values, cover_image: e.target.files.FileList });
          }}
        />
        <Card className="p-4">
          <label for="coverImage">
            <h6>Upload Cover Image</h6>
          </label>
        </Card>
      </Form.Group>
      <Form.Group>
        <Form.File
          id="coverImage"
          label="Custom file input"
          type="file"
          custom
          className="d-none"
          value={values.logo_img}
          onChange={handleChange("logo_img")}
        />
        <Card className="p-4">
          <label for="coverImage">
            <h6>Upload Logo Image</h6>
          </label>
        </Card>
      </Form.Group>
      <Form.Group className="d-flex justify-content-end">
        <Button
          onClick={handleStep}
          form="register-form"
          className="btn btn-primary mt-3"
          style={{ borderRadius: "20px" }}
        >
          Continue
        </Button>
      </Form.Group>
    </div>
  );
};

export default OutletDetails;
