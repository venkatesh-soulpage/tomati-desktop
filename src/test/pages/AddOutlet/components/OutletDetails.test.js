import React from "react";
import { render } from "@testing-library/react";
import OutletDetails from "pages/AddOutlet/components/OutletDetails";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("OutletDetails page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <OutletDetails handleChange={() => {}} handleFile={() => {}} />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
