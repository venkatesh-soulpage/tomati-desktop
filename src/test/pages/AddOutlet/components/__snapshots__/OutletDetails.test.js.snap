// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutletDetails page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="form-group"
    >
      <input
        class="form-control"
        placeholder="Menu Name"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <input
        class="form-control"
        placeholder="Phone Number"
        required=""
        type="number"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <textarea
        class="form-control"
        placeholder="Address"
        required=""
        rows="3"
        type="text"
      />
    </div>
    <div
      class="form-group"
    >
      <textarea
        class="form-control"
        maxlength="144"
        placeholder="Menu Bio"
        required=""
        rows="3"
        type="text"
      />
    </div>
    <div
      class="form-group"
    >
      <input
        accept=".png, .jpg,.jpeg"
        class="d-none custom-file"
        id="coverImage"
        label="Custom file input"
        type="file"
      />
      <div
        class="p-2 pt-4 b1-dash cr-p card"
      >
        <label
          class="cr-p"
          for="coverImage"
        >
          <h6>
             
            <img
              alt="icon"
              class="mx-4"
              src="UploadCover.svg"
            />
            <span>
              Upload Menu Cover photo
            </span>
          </h6>
        </label>
      </div>
    </div>
    <div
      class="form-group"
    >
      <div
        class="d-none custom custom-file"
      >
        <input
          accept=".png, .jpg,.jpeg"
          class="custom-file-input"
          id="logoImage"
          type="file"
        />
        <label
          class="custom-file-label"
          for="logoImage"
        >
          Custom file input
        </label>
      </div>
      <div
        class="p-2 d-flex pt-4 b1-dash cr-p card"
      >
        <label
          class="cr-p"
          for="logoImage"
        >
          <h6>
            <img
              alt="icon"
              class="mx-4"
              src="UploadCover.svg"
            />
            <span>
              Upload Logo
            </span>
          </h6>
        </label>
      </div>
    </div>
    <div
      class="d-flex justify-content-end form-group"
    >
      <button
        class="btn btn-primary mt-3 rounded-pill px-4 py-3 btn btn-primary"
        form="register-form"
        type="submit"
      >
        Continue
      </button>
    </div>
  </div>
</DocumentFragment>
`;
